libherokubuildpack/src/buildpack_output/mod.rs

```diff
@@ -0,0 +1,500 @@
+//! # Buildpack output
+//!
+//! Use the [`BuildpackOutput`] to output structured text as a buildpack is executing.
+//!
+//! ```
+//! use libherokubuildpack::buildpack_output::BuildpackOutput;
+//!
+//! let mut output = BuildpackOutput::new(std::io::stdout())
+//!     .start("Heroku Ruby Buildpack")
+//!     .warning("No Gemfile.lock found");
+//!
+//! output = output
+//!     .section("Ruby version")
+//!     .finish();
+//!
+//! output.finish();
+//! ```
+//!
+use crate::buildpack_output::ansi_escape::{BOLD_PURPLE, CYAN, RED, YELLOW};
+use crate::buildpack_output::util::{prefix_first_rest_lines, prefix_lines, ParagraphInspectWrite};
+use crate::write::line_mapped;
+use std::fmt::Debug;
+use std::io::Write;
+use std::time::Instant;
+
+mod ansi_escape;
+mod duration_format;
+pub mod style;
+mod util;
+
+/// See the module docs for example usage.
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug)]
+pub struct BuildpackOutput<T> {
+    pub(crate) started: Option<Instant>,
+    pub(crate) state: T,
+}
+
+/// Various states for [`BuildpackOutput`] to contain.
+///
+/// The [`BuildpackOutput`] struct acts as an output state machine. These structs
+/// are meant to represent those states.
+#[doc(hidden)]
+pub mod state {
+    use crate::buildpack_output::util::ParagraphInspectWrite;
+    use crate::write::MappedWrite;
+    use std::time::Instant;
+
+    #[derive(Debug)]
+    pub struct NotStarted<W> {
+        pub(crate) write: ParagraphInspectWrite<W>,
+    }
+
+    #[derive(Debug)]
+    pub struct Started<W> {
+        pub(crate) write: ParagraphInspectWrite<W>,
+    }
+
+    #[derive(Debug)]
+    pub struct Section<W> {
+        pub(crate) write: ParagraphInspectWrite<W>,
+    }
+
+    pub struct Stream<W: std::io::Write> {
+        pub(crate) started: Instant,
+        pub(crate) write: MappedWrite<ParagraphInspectWrite<W>>,
+    }
+}
+
+#[doc(hidden)]
```

Comment:
The `AnnounceSupportedState` trait is private, so I don't think this `#[doc(hidden)]` is needed?

```suggestion
```

Reply:
I had an issue with it that I mentioned in the commit https://github.com/heroku/libcnb.rs/commit/b9f6e2fdf8fcf0bf9d61480899912afce869e7bc. 

Though now when I delete the line, it is indeed fine:

```
$ git diff
diff --git a/libherokubuildpack/src/buildpack_output/mod.rs b/libherokubuildpack/src/buildpack_output/mod.rs
index d9c43dc..03e229b 100644
--- a/libherokubuildpack/src/buildpack_output/mod.rs
+++ b/libherokubuildpack/src/buildpack_output/mod.rs
@@ -160,7 +160,6 @@ pub mod state {
     }
 }
 
-#[doc(hidden)]
 trait AnnounceSupportedState {
     type Inner: Write;
 
⛄️ 3.1.4 🚀 /Users/rschneeman/Documents/projects/work/libcnb.rs (schneems/output)
$ cargo clippy --all-targets --all-features -- --deny warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.11s
```

I'm not 100% certain I know what caused the original problem or what change Manuel (or I) made that fixed it.